### ISSUE TITLE
Everyone get's admin options...

### ISSUE THREAD

**Issue by danieljwestman** [directus/next#535]
Directus v9.0.0-beta.9
PG 12

Every user get's the "Admin Options" in the account page. Should only be the one's with Admin Access I assume (and hope)... Only an admin should be able to choose role and set status... Correct?

To differens browsers open with different logins. The one on the left is a "editor user" checking out his account settings and the one the right is an "admin user" looking at the "editor role" settings. 

![Skärmavbild 2020-10-06 kl  10 04 47](https://user-images.githubusercontent.com/20950568/95175734-35099a80-07bc-11eb-8f24-b9d3a39f708d.png)

**Comment by maartenvn** [comment]
This has nothing to do with admin options you have not enabled in the role settings. I think this means Options you have as an admin to edit that user (eg: change role, ...)

**Comment by maartenvn** [comment]
However, those options indeed are displayed in the App when the user without admin privileges logs in.

**Comment by danieljwestman** [comment]
@maartenvn I'm not following you... As a logged in "none admin user" you shouldn't be able to change your own role, right? As it is now you can do that.

**Comment by maartenvn** [comment]
Yeah, you are right, it's an app issue since you aren't able tot submit the changes.

The API does provide a list of roles which should not be allowed

**Comment by danieljwestman** [comment]
Yup, it's some serious problems with permissions. 

It's also impossible to hide the Users from a none admin user. If you uncheck the Directus Users permissions for the "Editor role" that user doesn't have access to the App at all...

![Screenshot of Directus (3)](https://user-images.githubusercontent.com/20950568/95193479-2d55f000-07d4-11eb-9681-1115159239b2.jpg)

![Skärmavbild 2020-10-06 kl  13 02 02](https://user-images.githubusercontent.com/20950568/95193510-3646c180-07d4-11eb-948b-8255c36342ef.png)

**Comment by benhaynes** [comment]
Being able to limit permissions to "Directus Users" for Roles with App Access is not possible at the moment. The App needs certain information to function properly. You can remove the User Directory from the module bar, but this is a "soft" permission... and just hides it.

As for the "Admin Options", the API should block you from changing anything you don't have permission to (eg: changing your own role), and the plan is to eventually hide this section for everyone except admins... and hide User Preferences for everyone except the current user and admins. So this is less of a bug, and more of an improvement.

**Comment by danieljwestman** [comment]
Ouch, thats a big problem for us.

1) Showing the admin options (even if it can’t be saved) will misslead and our non tech users. And doesn’t look that good.
2) Showing all users to everyone is in conflict with GDPR and privacy for our users who’s not admins. Even if you can ”hide” it from the module bar, they appear when clicking on your avatar/account settings...

Possible to give some estamation for a fix on his? We need to be able to test our application/platform live pretty soon...

**Comment by rijkvanzanten** [comment]
> Being able to limit permissions to "Directus Users" for Roles with App Access is not possible at the moment. The App needs certain information to function properly. You can remove the User Directory from the module bar, but this is a "soft" permission... and just hides it.

The user needs permissions for it's own user, but you can set the permissions to `id` `_eq` `$CURRENT_USER` or `id` `_eq` `$CURRENT_ROLE` to only allow the user to see themselves or users in their role respectively

---

As for the hiding those fields on the users detail page based on access, the plan is to implement grouped fields first, and then hide the whole group as a unit. 

---

> Possible to give some ETA for a fix on his?

No hard ETA for that last thing. It's a new feature that isn't scheduled for the next release. If you want, you could already manually configure the permissions to limit read permissions to those explicit fields for your user roles.

> We need to be able to test our application/platform live pretty soon... 

Do keep in mind that you're using pre-release software that is explicitly marked not ready for production yet. If you have a need for a specific bugfix/feature, consider sponsoring the bug/feature to prioritize it and expedite the timeline

**Comment by danieljwestman** [comment]
Thanks @rijkvanzanten & @benhaynes for your great answers in the matter.

I’ll check on the ”work-arounds” that are suggested.

**Comment by danieljwestman** [comment]
@rijkvanzanten which variables do I have to work with in the permissions object?

**Comment by benhaynes** [comment]
https://github.com/directus/next/blob/main/docs/guides/users-and-roles.md#configuring-role-permissions
https://github.com/directus/next/blob/main/docs/reference/filter-rules.md
https://github.com/directus/next/blob/main/docs/reference/item-objects.md
